github/main.go

```diff
@@ -0,0 +1,97 @@
+package github
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/mholt/archiver"
+)
+
+type Release struct {
+	Version string `json:"tag_name"`
+	ZipUrl  string `json:"zipball_url"`
+}
+
+func DownloadLatestRelease(repo string, path string, dest string) string {
+	release := FetchLatestRelease(repo)
+
+	os.Chdir(path)
+	archivePath := fmt.Sprintf("%s.zip", release.Version)
+
+	err := DownloadFile(archivePath, release.ZipUrl)
+	defer os.Remove(archivePath)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if err := archiver.Unarchive(archivePath, path); err != nil {
+		log.Fatal(err)
+	}
+
+	org := strings.Split(repo, "/")[0]
+	dirs, _ := filepath.Glob(fmt.Sprintf("%s-*", org))
+
+	if len(dirs) == 0 {
+		log.Fatalln("Error: extracted release zip did not contain the expected directory")
+	}
+
+	for _, dir := range dirs {
+		err := os.Rename(dir, dest)
+
+		if err != nil {
+			os.RemoveAll(dir)
+			log.Fatal(err)
+		}
+	}
+
+	return release.Version
+}
+
+func FetchLatestRelease(repo string) Release {
```

Comment:
Can we provide a way to keep repo (e.g: `roots/trellis`) git history?

Reply:
🤔 that's a tough one. I guess it's possible to use the same API to get the release `tag_name`, then do a Git clone (via https?).. and then? `git clone --branch <tag_name> <repo_url>` is supported in newer Gits.